File: app/src/main/java/com/booboot/vndbandroid/model/vndbandroid/CacheItem.java
package com.booboot.vndbandroid.model.vndbandroid;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * Created by od on 12/03/2016.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public abstract class CacheItem {
    protected int vn;
    protected int added;

    public CacheItem() {
    }

    public CacheItem(int vn, int added) {
        this.vn = vn;
        this.added = added;
    }

    public int getVn() {
        return vn;
    }

    public void setVn(int vn) {
        this.vn = vn;
    }

    public int getAdded() {
        return added;
    }

    public void setAdded(int added) {
        this.added = added;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        CacheItem cacheItem = (CacheItem) o;
        return vn == cacheItem.vn;
    }

    @Override
    public int hashCode() {
        return vn;
    }
}

File: app/src/main/java/com/booboot/vndbandroid/model/vndbandroid/CacheItem.kt
package com.booboot.vndbandroid.model.vndbandroid

abstract class CacheItem {
    var vn: Int = 0
    var added: Int = 0

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val cacheItem = other as CacheItem?
        return vn == cacheItem!!.vn
    }

    override fun hashCode(): Int {
        return vn
    }
}